modules/nf-core/tabix/bgzip/tests/main.nf.test

```diff
@@ -0,0 +1,95 @@
+nextflow_process {
+
+    name "Test Process TABIX_BGZIP"
+    script "modules/nf-core/tabix/bgzip/main.nf"
+    process "TABIX_BGZIP"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "tabix"
+    tag "tabix/bgzip"
+
+    test("sarscov2_vcf_bgzip_compress") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                                [ id:'test' ],
+                                [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
+    test("homo_genome_bedgz_compress") {
+        when {
+            process {
+                """
+                input[0] = [
+                            [ id:'test' ],
+                            [ file(params.test_data['homo_sapiens']['genome']['genome_bed_gz'], checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
+    test("sarscov2_vcf_bgzip_compress_gzi") {
+        config "./bgzip_compress.config"
+        when {
+            process {
+                """
+                input[0] = [
+                                [ id:'test' ],
+                                [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
+    test("sarscov2_vcf_bgzip_compress_stub") {
+        config "./vcf_none.config"
+        options '-stub'
+        when {
+            process {
+                """
+                input[0] = [
+                                [ id:'test' ],
+                                [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(process.out).match() }
```

Comment:
There currently is a bug in `nf-test` that fails when you `touch` a file ending in `.gz`. The tool thinks it's gzipped because of this, which is not the case and it fails on that. You have two possible solutions:
1. You change the `touch ${prefix}.gz` command in the stub to `echo "" | gzip > ${prefix}.gz`
2. You only check the filenames with `nf-test` like this:
```
snapshot(
   file(process.out[0][1]).name
).match("stub_module")
```
(You should add the `file(...).name` code for every file in the output channel)

Reply:
can confirm this is a bug.....thanks @nvnieuwk

Reply:
Thanks both for the input. I can now get most of the testing working, tuples in the format `tuple val(meta), path("*.gz"), path("*.tbi")` are evading me for the bgziptabix module, I'm wondering whether the format should be changed to 2 channels in place of this 1.

Anywhoo...

Looks like the only major issue now is conda.

Where it gets interesting is that tabix 1.11 is defined everywhere, and used nowhere. 1.11 is the only version on Galaxy and Conda too as the only post-version 1 release.

I can only assume this is a change that was implemented after the original modules were made but there is no evidence of that.

Containers using 1.12 and conda using 1.19. There's also some small change in the output causing a change in the d5sum, however, I've unzipped a couple of the output files and they are visually identical. Index files between versions are identical so I'm guessing a newline character is to blame.

I guess it's time to contact the dev?

Reply:
`tabix` was officially moved to `htslib` from version 1.12 and on (if I remember correctly). Maybe it's just time to use `htslib` instead?

Reply:
Oh I see, then the issue is the same that we had with samtools.
You can just add htslib to pin it to the proper version

Reply:
cf https://github.com/nf-core/modules/pull/4773
https://github.com/nf-core/modules/pull/4831